### PR TITLE
Docs: Add docs for story tags

### PR DESCRIPTION
**Pull request by kylegach**
## What I did

Documented the newly-official `tags` API in #26634

#### Manual testing

1. Follow the steps in the [contributing instructions](https://storybook.js.org/docs/react/contribute/new-snippets#preview-your-work) for this branch, `tags-docs`
2. Check the new/updated pages
	1. `http://localhost:8000/docs/writing-stories/tags`
	2. `http://localhost:8000/docs/writing-docs/autodocs`

### Documentation

- [x] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [x] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [x] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

File: docs/api/main-config-tags.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+## Default tags
+
+The following tags are available by default in your project:
+
+```js
+{
+  dev: { description: 'Rendered in the Storybook UI, but only in development mode. They do not appear in the sidebar in production mode.' },
+  docs: { description: 'Rendered in the Storybook UI, but only in the docs page. They do not appear in the sidebar.' },
+  test: { description: 'Not rendered in the Storybook UI, in either development or production mode.' },
+}
```
Comment (kylegach) [inline docs/api/main-config-tags.md:21]: Is there more involved to each definition? e.g. I've seen reference to a `hidden` property.

File: docs/writing-stories/tags.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+Tags allow you to control which stories are included in your Storybook.
+
+## Default tags
+
+By default, the following tags are available in your project:
+
+- `dev`: Stories tagged with `dev` are rendered in the Storybook UI, but only in development mode. They do not appear in the sidebar in [production mode](../sharing/publish-storybook.md#build-storybook-as-a-static-web-application).
+- `docs`: Stories tagged with `docs` are rendered in the Storybook UI, but only in the [docs page](../writing-docs/autodocs.md). They do not appear in the sidebar.
+- `test`: Stories tagged with `test` are not rendered in the Storybook UI, in either development or production mode.
```
Comment (kylegach) [inline docs/writing-stories/tags.md:13]: Is this complete and correct?

File: docs/writing-stories/tags.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+If you want to add tags to all stories in your project, you can apply them in your `preview.js|ts` file:
+
+```ts
+// .storybook/preview.ts
+import { Preview } from '@storybook/<your-framework>';
+
+const preview: Preview = {
+  // ...rest of preview
+  tags: ['dev'],
```
Comment (kylegach) [inline docs/writing-stories/tags.md:70]: I can't think of an actual use case for this ability beyond `autodocs`.

**Review by shilman — APPROVED** [review]
Great copy! Great examples! 💯

**Comment by shilman** [comment]
Note: let's wait to merge until the tags PR gets merged

**Comment by kylegach** [comment]
> Note: let's wait to merge until the tags PR gets merged

@shilman — Should I just update the target branch to yours?

**Comment by shilman** [comment]
@kylegach SGTM!

File: docs/writing-stories/tags.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+| `autodocs` | No                            | Stories tagged with `autodocs` will be included in the [docs page](../writing-docs/autodocs.md). If a CSF file does not contain at least one story tagged with `autodocs`, that component will not generate a docs page. |
+| `dev`      | Yes                           | Stories tagged with `dev` are rendered in Storybook's sidebar.                                                                                                                                                           |
+| `test`     | Yes                           | Stories tagged with `test` do not currently affect Storybook's UI, but can be used to filter the [test runner](../writing-tests/test-runner.md#run-tests-for-a-subset-of-stories).                                       |
+
+The `dev` and `test` tags are automatically, implicitly applied to every story in your Storybook project.
+
+## Applying tags
+
+A tag can be any string, either the [built-in tags](#built-in-tags) or custom tags of your own design. To apply tags to a story, assign an array of strings to the `tags` property. Tags may be applied at the project, component (meta), or story levels.
```
[the remark this answers is not in the document]
Comment (kylegach) [reply docs/writing-stories/tags.md]: @JReinhold — How's this?

[`34c0e17` (#26715)](https://github.com/storybookjs/storybook/pull/26715/commits/34c0e17bd47159c45d47d6fc65a7d8af9f110fdb)

**Pull request merged**